src/metadata/metadata-storage.ts

```diff
@@ -286,4 +310,16 @@ export class MetadataStorage {
     }
     return authorizedField.roles;
   }
+
+  private findClassExtensions(target: Function): ExtensionsMetadata {
```

Comment:
Yeah this was also one of my initial ideas, but I couldn't find a way to make it right to the compiler, e.g.:
```
extensions.filter(
        entry => entry.target === target && (!(entry.fieldName || entry.fieldName === fieldName),
)
```
is not correct because `entry` can be of a type `ExtensionsClassMetadata` (which doesn't have `fieldName` property).

So, the best way I can think of to abstract this for now is this: b19c4afede1dbc7c5a8de0eda240ee89cc2b4d39 but I'm not really happy with the `any` cast.

If you have a better idea please correct it 🙂

Reply:
I know, union of array is tricky in TS. Changed it 😉